### ISSUE TITLE
Specimens tab repeats the same specimen ID in multiple rows

### ISSUE THREAD

**Issue by theathorn** [DataBiosphere/azul#331]
See https://dev.explore.data.humancellatlas.org/.

The specimen count is 3 so there should be only 3 rows of data shown.

┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/AZUL-216)
┆Project Name: azul
┆Issue Number: AZUL-216

**Comment by achave11** [comment]
Should change aggregate that yields a value for specimens, which uses `biomatirial_id` and use `entity_id` instead. To have an accurate count for the specimens display.

The error is in the metadata, the data object is not following the unique identifier set for that property.  Which we know is not true, because it yields this bug.
```
"biomaterial_id": {
      "description": "A unique ID for the biomaterial.",
      "type": "string",
      "user_friendly": "Biomaterial ID"
    }
```

**Comment by hannes-ucsc** [comment]
@achave11  and I spent some time looking into the root cause for why the specimens and files summaries are `permanently` inconsistent. The fix is essentially to only use primary keys to aggregate over when producing entity counts. For example, the specimen count is currently driven by a field called `biomaterial_id` which is supposed to the a unique identifier for specimens, but in fact, we indexed bundles that contain distinct (based on `document_id`) specimens with the same `biomaterial_id`. The fix for the specimen count is to count distinct `document_id` values, not distinct `biomaterial_id` values. There is a similar problem for projects which currently aggregates over `project_short_name`.

Once we fix those issues, the files and specimens summaries will be eventually consistent. They can still be temporarily inconsistent due concurrent indexing processes. We would like to maintain two separate summary endpoints so we have a way of telling whether the entity-specific indices are inconsistent.

@NoopDog proposed a different solution, which was to have one summary endpoint that hits all indices in separate queries and cherry picks the authoritative index for a particular the summary element. I prefer this solution because it is more efficient and exposes permanently inconsistent indices.

@NoopDog, this means that you should use `/summary/specimens` together with `/specimens` on the Specimens tab and `/summary/files` together with `/files` on the Files tab.

**Comment by hannes-ucsc** [comment]
The discrepancy between the row count and the count in the summary is caused by a combination of a metadata issue and the fact that the summary (as currently displayed on the Specimens tab in the data browser) is driven by the files summary:

Bundle `408312c4-936d-4246-a4dc-57aa0ef0f1f2`, version `2018-09-21T200551.075228Z` contains the specimen with UUID (`provenance.document_id`) `d075d9d4-0677-4dd3-95fd-b5425e58917d` which lists `biomaterial_core.biomaterial_id` as `Q4_DEMO-sample_SAMN02797092`.

Bundle `fbd52199-43d0-4ef5-afa7-e75b09893120`, version `2018-09-21T224119.697019Z` contains the specimen with UUID (`provenance.document_id`) `5f92562b-5836-44ca-9ae0-79fd86a5785a` which also lists `biomaterial_core.biomaterial_id` as `Q4_DEMO-sample_SAMN02797092`.

In other words, two different specimens in two separate bundles refer to the same `biomaterial_id`.

I will file a ticket with Ingest about this. Once we merge #276, the Specimens summary entry will match the number of rows but there will be more than one row with the same value in the Specimen ID column (that column is backed by biomaterial_id).
